tmva/tmva/src/MethodBDT.cxx

```diff
@@ -408,6 +419,15 @@ void TMVA::MethodBDT::DeclareOptions()
       fSepTypeS = "GiniIndex";
    }
 
+   // #### Number of CPUs to use in fPool multithreading, if you want the user to be able to set this
+   // rather than use all of the available CPUs, let them know about this option in the User's Guide or tutorials
+   // I just used it to time the algorithm vs number of CPUs.
+   #ifndef R__USE_IMT
+   DeclareOptionRef(fNumCPUs = 1, "NumCPUs", "Number of CPUs to use in multithreading.");
```

Comment:
IIUC this should probably be renamed to fNumThreads and "NumThreads" to better reflect that we are using multithreading.

Reply:
Shouldn't this be working on parallel implicitly w/o the need of specifying explicitly the number of cpus?

Reply:
After further contemplation: Yes, it most probably should.

Can we use imt for our benchmarking as well? I.e. Can one specify the size of the thread pool?

Reply:
@dpiparo I mainly created options for nthreads so that I could do the bench marking studies. However, the option is defaulted to use all of the available CPUs. If we would like the user to be able to set the number of CPUs then let them know about the option and their choice will override the default. If not, just don't let them know about it and it will be set automatically. The way this is coded is flexible.

Anyways, I would argue to default the threading option to the desired behavior for an uninformed user, but allow an informed user to have greater control -- the current set up.

Reply:
@ashlaban fNumThreads and numThreads are indeed better names. In regard to your other comment, there are options to specify the size of the threadpool if I remember correctly, but I just had it set up to use the number of threads as the pool size. Maybe @xvallspl can comment.

Reply:
@acarnes Yes, suggested in here https://github.com/root-project/root/pull/723#discussion_r125260915 (there's also a couple more comments I would like to be addressed before merging)

Your user can set the number of threads in their application with ROOT::EnableImplicitMT(), and then your class would ideally rely on that value, by default as much as your computer has available.